invertedai/large/common.py

```diff
@@ -0,0 +1,74 @@
+from typing import List, Optional, Tuple
+from pydantic import BaseModel
+
+from invertedai.common import AgentAttributes, AgentState, RecurrentState, Point
+
+class Region(BaseModel):
+    """
+    A region in a map used to divide a large simulation into smaller parts.
+
+    See Also
+    --------
+    AgentAttributes
+    """
+
+    center: Optional[Point] = None #: The center of the region if such a concept is relevant (e.g. center of a square, center of a rectangle)
+    size: Optional[float] = None #: Side length of the region for the default interpretation of a region as a square
+    agent_states: Optional[List[AgentState]] = [] #: A list of existing agents within the region
+    agent_attributes: Optional[List[AgentAttributes]] = [] #: The attributes of agents that exist within the region or that will be initialized within the region
+    recurrent_states: Optional[List[RecurrentState]] = [] #: Recurrent states of the agents contained within the region
```

Comment:
Correct, if recurrent_states is not empty, it should be the same length as agent_states AND agent_attributes. The two uses cases that I can identify for this object are for passing as arguments to DRIVE and to INITIALIZE which have slightly different needs so it's hard to enforce a standard across the board, especially because INITIALIZE has many different operating modes (i.e. equal number states and attributes, more attributes than states, no states at all).

Do you think it'd be worthwhile to split these regions up? Or have some sort of data validation? E.g. if agent details are to be added, it should check that it is a valid configuration?

Reply:
Yes, I would perform some validation.